File: Day2/a.ts
import {readFileSync} from 'fs'

const input = readFileSync('./Day2/input.txt', 'utf-8');

const guideArray = input.split('\r\n')

let totalPoints = 0
guideArray.forEach((line) => {
	let opponent = line.substring(0, 1)
	let myPlay = line.substring(2, 3)
	
	switch (myPlay) {
		case 'X':
			totalPoints += 1
			if (opponent == 'A')
				totalPoints += 3
			if (opponent == 'C')
				totalPoints += 6
			break;
		case 'Y':
			totalPoints += 2
			if (opponent == 'A')
				totalPoints += 6
			if (opponent == 'B')
				totalPoints += 3
			break;
		case 'Z':
			totalPoints += 3
			if (opponent == 'B')
				totalPoints += 6
			if (opponent == 'C')
				totalPoints += 3
		break;

	}
})
console.log(totalPoints);

File: Day2/src/bin/a.rs
fn main() {
    
    let loss = 0;
    let draw = 3;
    let win = 6;

    let mut sum = 0;
    std::fs::read_to_string("./input.txt").expect("Error reading file").lines().for_each(|line| {
        let opponent = line.split(" ").nth(0).unwrap();
        let me = line.split(" ").nth(1).unwrap();

        match me {
            "X" => {
                match opponent {
                    "A" => sum += draw,
                    "B" => sum += loss,
                    "C" => sum += win,
                    _ => panic!("Invalid input")
                }
                sum += 1;
            },
            "Y" => {
                match opponent {
                    "A" => sum += win,
                    "B" => sum += draw,
                    "C" => sum += loss,
                    _ => panic!("Invalid input")
                }
                sum += 2;
            },
            "Z" => {
                match opponent {
                    "A" => sum += loss,
                    "B" => sum += win,
                    "C" => sum += draw,
                    _ => panic!("Invalid input")
                }
                sum += 3;
            },
            _ => panic!("Invalid input")
        }
    });

    println!("{}", sum);

}
